segment.go

```diff
@@ -210,9 +211,23 @@ func (s *Segment) loadConfig() error {
 
 	numDocsOffset := storedIndexOffset - 8
 	s.numDocs = binary.BigEndian.Uint64(s.mm[numDocsOffset : numDocsOffset+8])
+	s.bytesRead += (8*4 + 4*3)
 	return nil
 }
 
+// Implements the segment.DiskStatsReporter interface
+// Only the persistedSegment type implments the
+// interface, as the intention is to retrieve the bytes
+// read from the on-disk segment as part of the current
+// query.
+func (s *Segment) SetBytesRead(val uint64) {
```

Comment:
Now that I'm looking at this - feels a bit distasteful to allow another library to just overwrite this parameter.

Reply:
I agree with you @abhinavdangeti,  
@Thejas-bhat ,

Why do we need this API? This would be inherent task within the merge process.
Meaning the the merge process would take care of  tracking the prev_bytes from segments and carrying it further to the newly formed segment.

Wouldn't that suffice the requirement?

Reply:
hm.. it may not possible without making disk format changes.